src/components/customModal/CustomModal.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import {View, Text, TouchableOpacity, Modal} from 'react-native';
+
+import {styles} from './styles';
+
+const CustomModal: React.FC<CustomModalProps> = ({
+  children,
+  visible,
+  setVisible,
```

Comment:
simply pass function - `onCloseModal` to this component

child components should not be given access to state setters directly

this will also allow you to add more functionality on modal close if needed